spec/index.bs

```diff
@@ -1113,6 +1101,20 @@ Given an {{HTMLScriptElement}} (|script|), this algorithm performs the following
 
     If the algorithm threw an error, rethrow the error.
 
+## <dfn-abstract-op export>Validate attribute mutation</dfn> ## {#validate-attribute-mutation}
+Given {{Attr}} |attribute|, {{Element}} |element| and {{TrustedType}} or a string |newValue|, this algorithm performs the following steps:
+
+1. If |element| has an <a spec="webidl" lt="attribute">IDL attribute</a> that [=reflects=] a content attribute with |attribute|'s [=Attr/local name=], let |idlAttribute| be that IDL attribute. Otherwise, return stringified |newValue|.
```

Comment:
This seems super magical. It's also missing a namespace check.

We also can't say "stringified". That needs to be defined.

Reply:
> This seems super magical

This mirrors https://w3c.github.io/trusted-types/dist/spec/#dom-trustedtypepolicyfactory-getattributetype, in general we decided to encode the controls at IDL layer, so if not reflection, we'd have to use some other mapping mechanism. We could just have a table of attr names + elements ->  type instead, but it feels wrong to duplicate what's already in IDL. An advice would be a great help here.

> Missing a namespace check

The reflection is only [defined for namespace-less attributes](https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#reflected-content-attribute-name), not sure why.

> "stringified" needs to be defined

It is defined? All [TrustedType objects](https://w3c.github.io/trusted-types/dist/spec/#trusted-html) have [stringifiers](https://webidl.spec.whatwg.org/#idl-stringifiers). Do you mean we shoudl avoid the colloquial term "stringified" here and replace it with the explicit algorithm, e.g. branching on a type?

Reply:
Reflection is not in IDL however. That's only the case in (some?) browsers.

Reply:
And yeah, you need to define what stringification means. A naïve implementation could use it as an excuse to execute JavaScript, for instance. (Or maybe that's what you want?)

Reply:
Removed IDL reflection. The stringification is now defined, but needs to be applied throughout this spec in other places too. PTAL.